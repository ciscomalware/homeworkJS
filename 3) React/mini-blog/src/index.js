import './css/style.css';
import 'bootstrap/dist/css/bootstrap.min.css';

fetch('https://jsonplaceholder.typicode.com/posts ')
  .then((posts) => posts.json())
  .then((posts) => {
    let idArr = [];
    posts.forEach((post) => {
      const { title } = post;
      const titleText = document.createTextNode(title);
      const header = document.createElement('div');
      header.classList.add('col-8');
      header.append(titleText);
      const divRow = document.querySelector('.row');
      divRow.append(header);
    });
    fetch('https://jsonplaceholder.typicode.com/users')
      .then((users) => users.json())
      .then((users) => {
        users.forEach((user) => {
          posts.map((post) => {
            if (user.id === post.userId) {
              const divCol4 = document.createElement('div');
              divCol4.classList.add('col-4');
              const userName = document.createTextNode(user.name);
              divCol4.append(userName);
              
            }
          });
        });
      });
  })
  .catch((err) => {
    throw err;
  });
