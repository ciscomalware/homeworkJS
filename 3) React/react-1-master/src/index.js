import 'bootstrap/dist/css/bootstrap.min.css';
import './css/style.css';
import app from './app';

app();
