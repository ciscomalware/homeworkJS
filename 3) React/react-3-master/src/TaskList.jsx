import React from 'react';

export default class TaskList extends React.Component {
  render() {
    const liElements = this.props.tasks.map((el) => (
      <li className={el.state === 'finished' ? 'list-group-item bg-success' : 'list-group-item bg-primary'} key={el.id}>
        <h5>{el.state === 'finished' ? <s>{el.title}</s> : el.title}</h5>
        <p>{el.state === 'finished' ? <s>{el.body}</s> : el.body}</p>
      </li>));

    return (
        <ul className="list-group-flush">
            {liElements}
        </ul>
    );
  }
}
