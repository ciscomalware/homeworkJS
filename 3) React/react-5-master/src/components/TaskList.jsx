import React from 'react';

export default class TaskList extends React.Component {
  render() {
    return (
      <div className="row">
        {this.props.tasks.map((task) => (task.state === 'finish'
          ? <div className="col-3" key={task.id}>
            <div className="card m-2 text-white bg-primary">
              <div className="card-header">
                <h3 className="card-title">{task.title}</h3>
              </div>
              <div className="card-body">
                <p className="card-text">{task.body}</p>
              </div>
              <div className="card-footer">
                <button type="button" className="btn btn-outline-light" onClick={this.props.finish(task.id)}>Завершить задачу</button>
              </div>
            </div>
          </div>
          : <div className="col-3" key={task.id}>
            <div className="card m-2 text-white bg-danger" key={task.id}>
              <div className="card-header">
                <h3 className="card-title">{task.title}</h3>
              </div>
              <div className="card-body">
                <p className="card-text">{task.body}</p>
              </div>
              <div className="card-footer">
                <button type="button" className="btn btn-outline-light" onClick={this.props.delete(task.id)}>Удалить задачу</button>
              </div>
            </div>
          </div>))}
      </div>
    );
  }
}
