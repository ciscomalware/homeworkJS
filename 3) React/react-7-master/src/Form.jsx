import React from 'react';
import { connect } from 'react-redux';
import * as actionCreators from './actions';

class Form extends React.Component {
  constructor(props) {
    super(props);
    this.state = { text: '' };
  }

  onSubmit = (e) => {
    e.preventDefault();
    const { addPost } = this.props;
    addPost({ text: this.state.text });
    this.setState({ text: '' });
  }

  onChange = (e) => {
    this.setState({ text: e.target.value });
  }

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <div className="input-group">
          <input type="text" name="title" className="form-control" onChange={this.onChange} value={this.state.text} />
          <div className="input-group-append">
            <button type="submit" className="btn btn-primary">Опубликовать</button>
          </div>
        </div>
      </form>
    );
  }
}

const mapStateToProps = (state) => ({ posts: state.posts });

export default connect(mapStateToProps, actionCreators)(Form);
