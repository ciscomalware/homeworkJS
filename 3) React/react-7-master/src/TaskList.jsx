import React from 'react';
import { connect } from 'react-redux';
import * as actionCreators from './actions';

class TaskList extends React.Component {
  delete = (id) => () => {
    const { removePost } = this.props;
    removePost({ id });
  }

  render() {
    const { posts } = this.props;
    const postToArray = Object.values(posts);
    return (
      <ul className="list-group-flush">
        {postToArray.map((task) => (
          <li className="list-group-item" key={task.id}>
            <h3>{task.text}</h3>
          <button type="button" className="btn btn-danger" onClick={this.delete(task.id)}>Удалить</button>
        </li>
        ))}
      </ul>
    );
  }
}

const mapStateToProps = (state) => ({ posts: state.posts });

export default connect(mapStateToProps, actionCreators)(TaskList);
