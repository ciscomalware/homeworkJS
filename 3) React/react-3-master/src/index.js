import React from 'react';
import ReactDOM from 'react-dom';
import { uniqueId } from 'lodash';
import 'bootstrap/dist/css/bootstrap.min.css';
import './css/style.css';
import TaskList from './TaskList';

const tasks = [
  {
    id: uniqueId(), title: 'Learn React', body: 'I should learn react', state: 'current',
  },
  {
    id: uniqueId(), title: 'Learn JS', body: 'Yay!', state: 'finished',
  },
  {
    id: uniqueId(), title: 'Learn HTML', body: 'DONE!', state: 'finished',
  },
  {
    id: uniqueId(), title: 'Learn Redux', body: 'Should learn after React', state: 'current',
  },
];

ReactDOM.render(<TaskList tasks={tasks} />, document.getElementById('container'));
