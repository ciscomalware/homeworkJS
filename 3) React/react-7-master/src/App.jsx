import React from 'react';
import Form from './Form';
import TaskList from './TaskList';

class App extends React.Component {
  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-12">
            <Form />
          </div>
          <div className="col-12">
            <TaskList />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
