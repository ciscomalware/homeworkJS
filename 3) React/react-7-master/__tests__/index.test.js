import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import App from '../src/App';
import TaskList from '../src/TaskList';
import Form from '../src/Form';
import rootReducer from '../src/reducers';
import * as actions from '../src/actions';


configure({ adapter: new Adapter() });

test('App', () => {
  const store = createStore(rootReducer);

  const component = mount(
    <Provider store={store}>
      <App />,
    </Provider>
  );
  // store.dispatch(actions.addPost({ data: 'Hello!' }));

  expect(component.find('TaskList')).toHaveLength(1);
  expect(component.html()).toMatchSnapshot();
  expect(component.find('TaskList').prop('posts')).toEqual({});
  component.find('Form').find('input').simulate('change', { target: { value: 'Hello, World!' } });
  component.find('Form').find('form').simulate('submit');
  expect(component.find('TaskList').prop('posts')).toEqual({ 1: { id: "1", text: 'Hello, World!' } });
  expect(component.find('TaskList').prop('posts')).toEqual(store.getState().posts);
  component.find('Form').find('input').simulate('change', { target: { value: 'Test' } });
  component.find('Form').find('form').simulate('submit');
  expect(component.find('TaskList').prop('posts')).toEqual({
    1: { id: "1", text: 'Hello, World!' },
    2: { id: "2", text: 'Test' },
  });
  expect(component.find('TaskList').prop('posts')).toEqual(store.getState().posts);
  expect(component.html()).toMatchSnapshot();

});