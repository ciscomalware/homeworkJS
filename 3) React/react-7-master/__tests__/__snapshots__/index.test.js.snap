// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"input-group\\"><input type=\\"text\\" name=\\"title\\" class=\\"form-control\\" value=\\"\\"><div class=\\"input-group-append\\"><button type=\\"submit\\" class=\\"btn btn-primary\\">Опубликовать</button></div></div></form></div><div class=\\"col-12\\"><ul class=\\"list-group-flush\\"></ul></div></div></div>,"`;

exports[`App 2`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"input-group\\"><input type=\\"text\\" name=\\"title\\" class=\\"form-control\\" value=\\"\\"><div class=\\"input-group-append\\"><button type=\\"submit\\" class=\\"btn btn-primary\\">Опубликовать</button></div></div></form></div><div class=\\"col-12\\"><ul class=\\"list-group-flush\\"><li class=\\"list-group-item\\"><h3>Hello, World!</h3><button type=\\"button\\" class=\\"btn btn-danger\\">Удалить</button></li><li class=\\"list-group-item\\"><h3>Test</h3><button type=\\"button\\" class=\\"btn btn-danger\\">Удалить</button></li></ul></div></div></div>,"`;
